src/Compilers/Core/Portable/Emit/EditAndContinue/EmitBaseline.cs

```diff
@@ -63,9 +64,14 @@ public sealed class EmitBaseline
 
         internal sealed class MetadataSymbols
         {
-            public readonly IReadOnlyDictionary<AnonymousTypeKey, AnonymousTypeValue> AnonymousTypes;
+            // In VB, AnonymousTypes contains all anonymous types and delegates, and the other two dictionaries are empty.
+            // In C#, AnonymousTypes contains just anonymous types, SynthesizedDelegates contains anonymous delegates
+            // where the parameter types and return types are generic type arguments, and AnonymousDelegates contains
+            // anonymous delegates where at least one of the parameter types or return type is not a valid type argument.
 
+            public readonly IReadOnlyDictionary<AnonymousTypeKey, AnonymousTypeValue> AnonymousTypes;
             public readonly IReadOnlyDictionary<SynthesizedDelegateKey, SynthesizedDelegateValue> SynthesizedDelegates;
+            public readonly IReadOnlyDictionary<string, AnonymousTypeValue> AnonymousDelegates;
```

Comment:
>AnonymousDelegates [](http://example.com/codeflow?start=76&length=18)

I think the naming is confusing. ```SynthesizedDelegates``` are also anonymous. I think the names should reflect that. I.e. they are anonymous, but different kind of anonymous.

Reply:
I agree the names are confusing currently. Any suggestions for alternatives? How about `AnonymousDelegates` and `AnonymousDelegatesWithFixedTypes`?